src/server/data.rs

```diff
@@ -1,19 +1,30 @@
+use chrono::Utc;
 use hyper::StatusCode;
 use semaphore::protocol::Proof;
 use semaphore::Field;
 use serde::{Deserialize, Serialize};
 
-use crate::identity_tree::{Hash, InclusionProof, RootItem};
+use crate::identity_tree::{Hash, InclusionProof, ProcessedStatus, RootItem, Status};
 use crate::prover::{ProverConfig, ProverType};
 
 #[derive(Debug, Serialize, Deserialize, PartialEq)]
-pub struct InclusionProofResponse(pub InclusionProof);
+pub struct InclusionProofResponse {
```

Comment:
Isn't this type identitcal to `InclusionProof`?

Reply:
It is but I think it is beneficial to keep this two domains (API and internal tree) not mixed. It is much easier to not brake the API by accident as we have customers that rely on it. That's why I also changed this way VerifySemaphoreProofResponse. Now hiding status is also not called as a function that someone may miss or forget to use. It is always done when converting from VerifySemaphoreProof. For me it is also much cleaner now what is happening there and when. :) I've found it hard when reverting that status and trying to ensure we are consistent with previous behaviour used by other teams.

Reply:
Ok, sounds good